src/fable-library-py/fable_library/date.py

```diff
@@ -231,36 +276,224 @@ def add(x: datetime, y: TimeSpan) -> datetime:
+        # 9/10/2014 1:50:34
+        r"^(0?[1-9]|1[0-2])\/(0?[1-9]|1[0-2])\/\d{4} ([0-9]|(0|1)[0-9]|2[0-4]):([0-5][0-9]|0?[0-9]):([0-5][0-9]|0?[0-9])$": "%m/%d/%Y %H:%M:%S",
+        # Matches a datetime string in the format "YYYY-MM-DDTHH:MM:SS.ffffff". This format usually parses with
+        # `fromisoformat`, but not with Python 3.10
+        r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{7}$": "%Y-%m-%dT%H:%M:%S.%f000",
+    }
+
+    for pattern, format in formats.items():
+        if re.fullmatch(pattern, string):
+            return datetime.strptime(string, format)
+
+    raise ValueError("Unsupported format by Fable: %s" % (string))
+
+
+def try_parse(string: str, defValue: FSharpRef[datetime]) -> bool:
     try:
         defValue.contents = parse(string)
         return True
     except Exception:
         return False
 
 
+def date(d: datetime) -> datetime:
+    return create(d.year, d.month, d.day, 0, 0, 0, 0, kind(d))
+
+
+def is_leap_year(year: int) -> bool:
+    return year % 4 == 0 and (year % 100 != 0 or year % 400 == 0)
+
+
+def days_in_month(year: int, month: int) -> int:
+    if month == 2:
+        return 29 if is_leap_year(year) else 28
+
+    if month in (4, 6, 9, 11):
+        return 30
+
+    return 31
+
+
+def add_years(d: datetime, v: int) -> datetime:
+    new_month = month(d)
+    new_year = year(d) + v
+    _days_in_month = days_in_month(new_year, new_month)
+    new_day = min(_days_in_month, day(d))
+    return create(new_year, new_month, new_day, hour(d), minute(d), second(d), millisecond(d), kind(d))
+
+
+def add_months(d: datetime, v: int) -> datetime:
+    new_month = month(d) + v
+    new_month_ = 0
+    year_offset = 0
+    if new_month > 12:
+        new_month_ = int(fmod(new_month, 12))
+        year_offset = new_month // 12
+        new_month = new_month_
+    elif new_month < 1:
+        new_month_ = 12 + int(fmod(new_month, 12))
+        year_offset = new_month // 12 + (-1 if new_month_ == 12 else 0)
+        new_month = new_month_
+    new_year = year(d) + year_offset
+    _days_in_month = days_in_month(new_year, new_month)
+    new_day = min(_days_in_month, day(d))
+    return create(new_year, new_month, new_day, hour(d), minute(d), second(d), millisecond(d), kind(d))
+
+
+def add_days(d: datetime, v: int) -> datetime:
+    return d + timedelta(days=v)
+
+
+def add_hours(d: datetime, v: int) -> datetime:
+    return d + timedelta(hours=v)
+
+
+def add_minutes(d: datetime, v: int) -> datetime:
+    return d + timedelta(minutes=v)
+
+
+def add_seconds(d: datetime, v: int) -> datetime:
+    return d + timedelta(seconds=v)
+
+
 def add_milliseconds(d: datetime, v: int) -> datetime:
     return d + timedelta(milliseconds=v)
 
 
+def add_microseconds(d: datetime, v: int) -> datetime:
+    return d + timedelta(microseconds=v)
+
+
+def kind(d: datetime) -> DateKind:
+    if d.tzinfo == timezone.utc:
```

Comment:
I don't think this will be able to see the difference between localtime and utc if localtime = utc

Reply:
You are right